enhancements/monitoring/metrics-collection-profiles.md

```diff
@@ -0,0 +1,515 @@
+    Staleness occurs at 5 minutes (by default) which is would cause gaps in
+    graphs/dashboards;
+  - Many upstream dashboards were built while having the 30 second interval in
+    mind;
+  - Scrapes can fail, but the user might not be mindful of this and set a high
+    scrape interval;
+- Add a separate container to prometheus-operator (p-o) that would be used by
+  p-o to modify the prometheus config according to a metrics collection profile.
+  - This container would perform an analysis on what metrics were being used.
+    Then it would provide prometheus operator with this list.
+  - Prometheus-operator with the list provided by this new component would know
+    what scraping targets it could change to keep certain metrics.
+- Recently Azure also added support for metrics collection profiles:
+  - [Azure
+    Docs](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/prometheus-metrics-scrape-configuration-minimal)
+  - https://github.com/Azure/prometheus-collector
+  - In their approach they also have
+    [hardcoded](https://github.com/Azure/prometheus-collector/blob/66ed1a5a27781d7e7e3bb1771b11f1da25ffa79c/otelcollector/configmapparser/tomlparser-default-targets-metrics-keep-list.rb#L28)
+    set of metrics that are only consumed when the minimal profile is enabled.
+    However, customers are also able to extend this minimal profile with regexes
+    to include metrics which might be interesting to them.
+- Leverage [installer
+  capabilities](https://docs.google.com/document/d/1I-YT7LKKDHSBLB6Hmg0tZ54DWjrAxlVdXxlViShMu-0/edit#heading=h.848jsje80fru)
+  - After some consideration we decided to abandon this idea since it would only
+    work for resources controlled by CVO which is not the case for the majority
+    of ServiceMonitors.
+  - Additionally, this requires users to "commit" to one profile throughout the
+    cluster lifecycle, which is a bit static for our needs, for more details,
+    PTAL at [the reasoning
+    here](https://github.com/openshift/enhancements/pull/1298#issuecomment-1895513051).
+
+### Adopted metrics collection profiles
+
+Add the team and the component that will adopt metrics collection profiles
+and implementation status. Possible implementation status: 
+- considering
+- implementation in progress
+- implemented
+
+| Team            | Component          | Implementation Status      |
+|-----------------|--------------------|----------------------------|
+| Monitoring Team | kubelet            | Implemented                |
+| etcd Team       | etcd               | Implemented                |
+| Monitoring Team | kube-state-metrics | Implemented                |
+| Monitoring Team | node-exporter      | Implemented                |
+| Monitoring Team | prometheus-adapter | Implemented                |
+
+### Topology Considerations
+
+Supported on all topologies that deploy CMO.
+
+#### Hypershift / Hosted Control Planes
```

Comment:
It might be good detailing how it will look like for HCP clusters. In particular HyperShift controls the scrape configuration of control plane components so collection profile won't apply for them.